src/components/Personalization/createViewCacheManager.js

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { assign } from "../../utils";
+
+export default () => {
+  let viewStorage;
+
+  const storeViews = decisions => {
+    if (viewStorage === undefined) {
+      viewStorage = {};
+    }
+    assign(viewStorage, decisions);
+  };
+
+  const getView = viewName => {
+    return viewStorage[viewName];
+  };
+
+  const isInitialized = () => {
```

Comment:
This implementation doesn't take into account when the cache is initializing, i.e. a request has gone out to fetch decisions that will be put in the cache. If I were doing this, I'd setup the cache with promises like this:

```
export default () => {
  let viewStorage;
  let viewStorageDeferred = defer();

  const storeViews = decisionsPromise => {
    if (viewStorage === undefined) {
      viewStorage = {};
    }
    decisionsPromise.then((decisions) => {
      assign(viewStorage, decisions);
      viewStorageDeferred.resolve();
    })
  }

  const getView = viewName => {
    viewStorageDeferred.then(() => {
      return viewStorage[viewName];
    })
  };

  const isInitialized = () => {
    return !(viewStorage === undefined);
  }

  return {
    storeViews,
    getView,
    isInitialized
  };
};
```

So with this implementation, storeViews accepts a promise that should return the decisions to store. This should be called before the fetchDecisions request returns. Also the getView returns a promise so that if the cache is initializing you can wait for the first request to return.

Reply:
Could `storeViews` be called multiple times on the same page? I'm having a hard time figuring this out. If so, then we may need to think about this even more.